### PR TITLE
ELA-1118: Add schema for the filter settings

### LINKED ISSUE #411: Drupal 10.3: 'responsive' is not a supported key.
### The problem

Drupal 10.3 + oe_bootstrap_theme 

Error message:
`'responsive' is not a supported key.`

While saving an edit page of any text format using CKEditor 5, ex. http://web:8080/web/en/admin/config/content/formats/manage/full_html

This issue doesn't happen with Drupal 10.2 (downgrade fixes the issue).

Also, when removing below file, the problem is gone: https://github.com/openeuropa/oe_bootstrap_theme/blob/1.x/modules/oe_bootstrap_theme_helper/src/Plugin/Filter/BootstrapTable.php

### PR DESCRIPTION
**Pull request by piotrsmykaj**
Jira issue(s):
- [ELA-1118](https://webgate.ec.europa.eu/CITnet/jira/browse/ELA-1118)

GitHub issue:
- https://github.com/openeuropa/oe_bootstrap_theme/issues/411

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-version: '2'
```
Comment (AaronGilMartinez) [inline docker-compose.yml]: This change doesn't seem related with the issue. Why this?

File: modules/oe_bootstrap_theme_helper/src/Plugin/Filter/BootstrapTable.php [hunk 2]
```diff
@@ -6,21 +6,23 @@
...
- * )
  */
+#[Filter(
+  id: "oe_bootstrap_theme_table",
+  title: new TranslatableMarkup("Bootstrap tables"),
+  type: FilterInterface::TYPE_TRANSFORM_IRREVERSIBLE,
+  settings: [
+    "responsive" => "",
+  ],
+)]
```
Comment (AaronGilMartinez) [inline modules/oe_bootstrap_theme_helper/src/Plugin/Filter/BootstrapTable.php]: Core introduced discovery for filter Attributes on 10.3.0 https://www.drupal.org/node/3229001
This change won't support 10.2.x

**Review by AaronGilMartinez — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (piotrsmykaj) [reply docker-compose.yml]: while working on this fix I received local warnings about a deprecated version parameter

File: modules/oe_bootstrap_theme_helper/src/Plugin/Filter/BootstrapTable.php (hunk 2 above)
Comment (piotrsmykaj) [reply modules/oe_bootstrap_theme_helper/src/Plugin/Filter/BootstrapTable.php]: yes good point, in this case it needs to be reverted

File: docker-compose.yml (hunk 1 above)
Comment (AaronGilMartinez) [reply docker-compose.yml]: The "version" is used to declare file format it's only used for Compose v1, Compose v2 will display a warning and this setting be ignored. I know the intention is good, but this is out of scope and it may be used in some cases, lets remove this change.

**Review by AaronGilMartinez — DISMISSED** [review]

**Comment by julien-** [comment]
Take a look in https://www.drupal.org/project/drupal/releases/10.3.2

> Text formats could not be saved due to config translation validation new in Drupal 10.3. The LogicException has been replaced with a E_USER_DEPRECATED error.

**Review by brummbar — APPROVED** [review]
